### PR TITLE
Add regression test for issue #76042

### PR DESCRIPTION
**Pull request by LeSeulArtichaut**
Originally posted in https://github.com/rust-lang/rust/issues/76042#issuecomment-683272710.
r? @pnkfelix

### CODE REVIEW

File: src/test/ui/issues/issue-76042.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// run-pass
```
Comment (tmiasko) [inline src/test/ui/issues/issue-76042.rs:1]: I would expect that this that would have a few extra compile flags. `compile-flags: -Coverflow-checks=off -Ccodegen-units=1 -Copt-level=0` with `--target i686-unknown-linux-gnu` also being relevant but omitted.

**Comment by RalfJung** [comment]
(FWIW, I'd appreciate if you could create such branches in forks, not the main repo... it is somewhat confusing to have these things how up in my `git branch -r` or when navigating the main Rust repo.^^)

**Comment by LeSeulArtichaut** [comment]
Oops, sorry, I didn’t even notice. I just used GitHub UI to create the commit and forgot I had write access :)

File: src/test/ui/issues/issue-76042.rs (hunk 1 above)
Comment (LeSeulArtichaut) [reply src/test/ui/issues/issue-76042.rs:1]: I don't think `--target i686-unknown-linux-gnu` (or rather `only-i686-unknown-linux-gnu`) isn't useful here, this test should pass for every target.
I'll add `compile-flags: -Coverflow-checks=off -Ccodegen-units=1 -Copt-level=0` which I didn't think were needed for the repro. My bad

**Comment by pnkfelix** [comment]
r=me once the test is actually updated in the manner described in https://github.com/rust-lang/rust/pull/77151#discussion_r496193233

**Comment by LeSeulArtichaut** [comment]
@pnkfelix Updated the test

**Pull request merged**